compendium/LogService/include/cppmicroservices/logservice/Logger.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef CPPMICROSERVICES_LOGGER_H_
+#define CPPMICROSERVICES_LOGGER_H_
+
+#include "cppmicroservices/ServiceReferenceBase.h"
+
+#include <cstdint>
+#include <exception>
+#include <string>
+#include <functional>
+
+namespace sinks
+{
+    class sink;
+}
+
+namespace spdlog
+{
+    class logger;
+    using sink_ptr = std::shared_ptr<sinks::sink>;
+} // namespace spdlog
+
+namespace cppmicroservices::logservice
+{
+        /** @}*/
+
+        /**
+         * \ingroup MicroService
+         * \ingroup gr_logservice
+         *
+         * Provides methods for bundles to write messages to the log.
+         * Logger interface defines several methods for each of the defined LogLevels.
+         *
+         */
+
+	enum class LogLevel
+        {
+            audit, //This log level is used for information that must always be logged.
+ 
+            error, //This log level is used for information about an error situation.
+ 
+            warn, //This log level is used for information about a failure or unwanted situation that is not blocking.
+ 
+            info, //This log level is used for information about normal operation.
+ 
+            debug, //This log level is used for detailed output for debugging operations.
+ 
+            trace //This log level is used for large volume of output for tracing operations.   
+	};
+	
+        class Logger
+        {
+          public:
+
+            virtual ~Logger() = default;
+
+	    Logger() = default;
```

Comment:
this class is abstract and cannot be instantiated. There should not be a default constructor.